src/Symfony/Component/DependencyInjection/Exception/ArgumentCircularReferenceException.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Exception;
+
+/**
+ * This exception is thrown when a circular reference is detected between services in arguments.
+ *
+ * @author Nicolas LEFEVRE <weblefevre@gmail.com>
+ */
+class ArgumentCircularReferenceException extends RuntimeException
```

Comment:
if we want this to make it into a bug fix release, we should not add any new class in the PR (otherwise that'd be a smell for a new feature).
Instead, what about throwing a ServiceCircularReferenceException?

Reply:
OK, throwing `ServiceCircularReferenceException` should be a good idea...